kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/kroxylicious/KroxyliciousConfigMapTemplates.java

```diff
@@ -46,16 +53,40 @@ public static ConfigMapBuilder defaultKroxyliciousConfig(String clusterName, Str
      *
      * @param clusterName the cluster name
      * @param namespaceName the namespace name
-     * @param topicName the topic name
-     * @param config config
+     * @param testKmsFacade the test kms facade
      * @return the config map builder
      */
-    public static ConfigMapBuilder kroxyliciousRecordEncryptionConfig(String clusterName, String namespaceName, String topicName, Config config) {
+    public static ConfigMapBuilder kroxyliciousRecordEncryptionConfig(String clusterName, String namespaceName, TestKmsFacade<?, ?, ?> testKmsFacade) {
         return baseKroxyliciousConfig(namespaceName)
-                .addToData("config.yaml", getRecordEncryptionConfigMap(clusterName, topicName, config));
+                .addToData("config.yaml", getRecordEncryptionConfigMap(clusterName, testKmsFacade));
+    }
+
+    private static String buildEncryptionFilter(TestKmsFacade<?, ?, ?> testKmsFacade) {
+        return "- type: RecordEncryption"
+                + "\n  config:"
+                + "\n    kms: " + testKmsFacade.getKmsServiceClass().getSimpleName()
+                + "\n    kmsConfig:"
+                + "\n      " + getYamlKmsConfig(testKmsFacade.getKmsServiceConfig())
+                + "\n    selector: TemplateKekSelector"
+                + "\n    selectorConfig:"
+                + "\n      template: \"${topicName}\"";
     }
 
-    private static String getRecordEncryptionConfigMap(String clusterName, String topicName, Config config) {
+    private static String getYamlKmsConfig(Object config) {
+        String configYaml;
+        try {
+            configYaml = YAML_OBJECT_MAPPER.writeValueAsString(config).replace("---", "").indent(6).trim();
```

Comment:
The replace and the fixed indent both feel a little off. I suspect we can tune `YAML_OBJECT_MAPPER` to avoid the need for the replace.
Switching to a text block might remove the need to call indent. 

Maybe refactoring to `getNestedYaml(Object config, int indent)` might be useful move to make clear the expected usage.